features/support/world.js

```diff
@@ -0,0 +1,136 @@
+'use strict';
+
+const Assert = require('chai').assert;
+const path = require('path');
+const env = require('node-env-file');
+const requestretry = require('requestretry');
+const request = require('../support/request');
+const { defineSupportCode } = require('cucumber');
+
+/**
+ * Retry until the build has finished
+ * @method retryStrategy
+ * @param  {Object}      err
+ * @param  {Object}      response
+ * @param  {Object}      body
+ * @return {Boolean}     Retry the build or not
+ */
+function buildRetryStrategy(err, response, body) {
+    return err || body.status === 'QUEUED' || body.status === 'RUNNING';
+}
+
+/**
+ * Promise to wait a certain number of seconds
+ * @method promiseToWait
+ * @param  {Number}      timeToWait  Number of seconds to wait before continuing the chain
+ * @return {Promise}
+ */
+function promiseToWait(timeToWait) {
+    return new Promise((resolve) => {
+        setTimeout(() => resolve(), timeToWait * 1000);
+    });
+}
+
+/**
+ * Ensure a pipeline exists, and get its jobs
+ * @method ensurePipelineExists
+ * @param   {Object}    config
+ * @param   {String}    config.repoName     Name of the pipeline
+ * @return {Promise}
+ */
+function ensurePipelineExists(config) {
```

Comment:
It's used in both `secrets` and `events`, so if the regex does get changed to match both maybe there should be some sort of file like `shared-steps`